### PR TITLE
Global variables checker dev

### PR DESCRIPTION
**Pull request by nigef**
The `global_variables_checker` would think variables inside a list comprehension are always global -- these variables should never be seen as global, purely from being in a list comp.

### CODE REVIEW

File: python_ta/checkers/global_variables_checker.py [hunk 1]
```diff
@@ -76,6 +76,8 @@ def _get_child_disallowed_global_var_nodes(node):
     global, non-constant variable. 
     """
     node_list = []
+    if isinstance(node.parent, astroid.ListComp) or isinstance(node.parent, astroid.Comprehension):
```
Comment (david-yz-liu) [inline python_ta/checkers/global_variables_checker.py]: Two comments here:

1. If `ListComp` is necessary, then there should also be a check for `DictComp` and `SetComp` (there's a symmetry here).
2. But I think `node.parent` is the wrong approach (doesn't take into account nested expressions). I'd like to see you use `lookup_scope` (or whatever that method was called) to do this check.

Comment (nigef) [inline python_ta/checkers/global_variables_checker.py]: Thanks, I've added the checks for additional nodes.
Nice point about the nesting, although I can't think of a case where this doesn't work on nested expressions, so maybe the current approach is acceptable?

Comment (david-yz-liu) [inline python_ta/checkers/global_variables_checker.py]: About the nesting: something like `[x + 3 for x in ...]` would have the `x` in `x+3` as a child of a `BinOp` node, right? If I understand this code properly, in this case the `x` in `x + 3` would be reported.

Comment (nigef) [inline python_ta/checkers/global_variables_checker.py]: Fixed

**Pull request closed without merging**